dotnet-interactive/JupyterKernelSpecInstaller.cs

```diff
@@ -2,77 +2,155 @@
         {
-            _kernelSpecModule = new JupyterKernelSpecModule();
         }
-        public async Task<KernelSpecInstallResult> InstallKernel(DirectoryInfo sourceDirectory, DirectoryInfo destination = null)
+
+        public JupyterKernelSpecInstaller(IConsole console, IJupyterKernelSpecModule jupyterKernelSpecModule)
+        {
+            _console = console ?? throw new ArgumentNullException(nameof(console));
+            _kernelSpecModule = jupyterKernelSpecModule;
+        }
+
+
+        public async Task<bool> InstallKernel(DirectoryInfo sourceDirectory, DirectoryInfo destination = null)
         {
+            var kernelDisplayName = GetKernelDisplayName(sourceDirectory);
 
             if (destination != null)
             {
-                var (succeeded, message) = CopyKernelSpecFiles(sourceDirectory, destination);
-                return new KernelSpecInstallResult(succeeded, message);
+                return InstallKernelSpecToDirectory(sourceDirectory, destination, kernelDisplayName);
             }
-            else
+
+            try
             {
-                try
+                var result = await _kernelSpecModule.InstallKernel(sourceDirectory);
+                if (result.ExitCode == 0)
                 {
-                    var result = await _kernelSpecModule.InstallKernel(sourceDirectory);
-                    if (result.ExitCode == 0)
-                    {
-                        return new KernelSpecInstallResult(true,
-                            string.Join('\n', result.Output.Concat(result.Error)));
-                    }
+                    _console.Out.WriteLine($"Installed \"{kernelDisplayName}\" kernel.");
+                    return true;
                 }
-                catch (Win32Exception w32e)
+            }
+            catch (Win32Exception w32e)
+            {
+                // file not found when executing process
+                if (!w32e.Source.Contains(typeof(System.Diagnostics.Process).FullName))
                 {
-                    // file not found when executing process
-                    if (!w32e.Source.Contains(typeof(System.Diagnostics.Process).FullName))
-                    {
-                        throw;
-                    }
+                    _console.Error.WriteLine($"Failed installing \"{kernelDisplayName}\" kernel.");
+                    throw;
                 }
+            }
 
-                var notAvailable = "The kernelspec module is not available.";
+            destination = _kernelSpecModule.GetDefaultKernelSpecDirectory();
 
-                destination = GetDefaultDirectory();
-                
-                if (!destination.Exists)
-                {
-                    return new  KernelSpecInstallResult(false, string.Join('\n', notAvailable, $"The kernelspec path ${destination.FullName} does not exist."));
-                }
+            _console.Out.WriteLine("The kernelspec module is not available.");
```

Comment:
This reads like an error. What is the user supposed to do with this information?

Reply:
so should that go to the error stream?

Reply:
If it's an error, but it doesn't seem like one, in context. It just reads like one.

Is it useful info for the user?